src/main/kotlin/io/github/mmarco94/compose/adw/components/Dialog.kt

```
@@ -168,4 +170,64 @@ fun AboutDialog(
+    val label: String,
+    val appearance: ResponseAppearance = ResponseAppearance.DEFAULT,
+    val isEnabled: Boolean = true,
+)
+
+@Composable
+fun AlertDialog(
+    modifier: Modifier = Modifier,
+    contentHeight: Int = -1,
+    contentWidth: Int = -1,
+    followsContentSize: Boolean = false,
+    presentationMode: DialogPresentationMode = DialogPresentationMode.AUTO,
+    title: String? = null,
+    onClose: () -> Unit = {},
+    heading: String,
+    body: String,
+    responses: List<AlertDialogResponse>,
+    defaultResponse: AlertDialogResponse? = null,
+    onResponse: (AlertDialogResponse) -> Unit,
+) {
+    var connection: SignalConnection<*>? by remember { mutableStateOf(null) }
+
+    BaseDialog(
+        modifier = modifier,
+        contentHeight = contentHeight,
+        contentWidth = contentWidth,
+        followsContentSize = followsContentSize,
+        presentationMode = presentationMode,
+        title = title,
+        onClose = onClose,
+        creator = {
+            AlertDialog.builder().build()
+        },
+        updater = {
+            set(heading) { this.widget.heading = it }
+            set(body) { this.widget.body = it }
+            set(responses) {
+                responses.forEach { response ->
+                    if (!this.widget.hasResponse(response.id)) {
+                        this.widget.addResponse(response.id, response.label)
+                    }
+                    this.widget.setResponseEnabled(response.id, response.isEnabled)
+                    this.widget.setResponseAppearance(response.id, response.appearance)
+                }
+            }
+            set(defaultResponse) { this.widget.defaultResponse = it?.id }
+            set(onResponse) {
+                connection?.disconnect()
+                connection = this.widget.onResponse(null) { responseId ->
+                    if (responseId == "close") return@onResponse
+                    responses.firstOrNull { it.id == responseId }
```

Minor, non-blocking: Should we be more ambitious, and use `single` instead of `firstOrNull`?

GTK generally fails open on errors, so maybe you're right, and ignoring errors is the right approach. Should we at least log a warning?

---

PS: I've added a logging framework on https://github.com/MMarco94/compose-4-gtk/commit/5f028f05cecf2b8bf5ee4de4e9536990c2a0ac25#diff-1ae823fcd1cfb976da8b4032656efc82f19a7abafb4f6ef9271a680d6dd3f9c5 if you want to see how to add a log

---

Left lenient behavior, but added a log.